pkg/cmd/webapp/init/init.go

```diff
@@ -115,22 +113,38 @@ func NewCmd(f *cmdutil.Factory) *cobra.Command {
 	return NewCobraCmd(NewInitCmd(f))
 }
 
-func (cmd *InitCmd) run(info *InitInfo, options *contracts.AzionApplicationOptions, initCmd *cobra.Command) error {
+func (cmd *InitCmd) run(info *InitInfo, options *contracts.AzionApplicationOptions, c *cobra.Command) error {
 	if info.YesOption && info.NoOption {
 		return msg.ErrorYesAndNoOptions
 	}
 
-	//gets the test function (if it could not find it, it means it is currently not supported)
-	testFunc, ok := makeTestFuncMap(cmd.Stat)[info.TypeLang]
-	if !ok {
-		return utils.ErrorUnsupportedType
+	path, err := cmd.GetWorkDir()
+	if err != nil {
+		return err
 	}
 
-	path, err := cmd.GetWorkDir()
+	bytePackageJson, pathPackageJson, err := ReadPackageJson(cmd, path)
 	if err != nil {
 		return err
 	}
 
+	projectName, projectSettings, err := DetectedProjectJS(bytePackageJson)
+	if err != nil {
+		return err
+	}
+	fmt.Fprintf(cmd.Io.Out, msg.WebappAutoDetectec, projectSettings) // nolint:all
+
+	if !hasThisFlag(c, "type") {
```

Comment:
If the function does not do anything besides just checking for the flag, might as well check the flag directly here.